setup.py

```diff
@@ -1,239 +1,233 @@
-import warnings
 import os
 import re
 import ast
+from collections import namedtuple
 from pathlib import Path
-from packaging.version import parse, Version
-import platform
+from typing import Dict
+from shutil import which
+from packaging.version import Version, parse
 
-from setuptools import setup, find_packages
 import subprocess
 
-import urllib.request
-import urllib.error
-from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
-
 import torch
 from torch.utils.cpp_extension import (
     BuildExtension,
-    CppExtension,
-    CUDAExtension,
     CUDA_HOME,
 )
 
+from setuptools import Extension, find_packages, setup
+from setuptools.command.build_ext import build_ext
+
+logger = logging.getLogger(__name__)
+
+# Enivronment variables
+Envs = namedtuple("Envs", ["VERBOSE", "MAX_JOBS", "NVCC_THREADS", "VLLM_TARGET_DEVICE", "CMAKE_BUILD_TYPE"])
+envs = Envs(
+    VERBOSE=bool(int(os.getenv("VERBOSE", "0"))),
+    MAX_JOBS=os.getenv("MAX_JOBS"),
+    NVCC_THREADS=os.getenv("NVCC_THREADS"),
+    VLLM_TARGET_DEVICE=os.getenv("VLLM_TARGET_DEVICE", "cuda"),
+    CMAKE_BUILD_TYPE=os.getenv("CMAKE_BUILD_TYPE"),
+)
 
 with open("README.md", "r", encoding="utf-8") as fh:
     long_description = fh.read()
 
-
 # ninja build does not work unless include_dirs are abs path
 this_dir = os.path.dirname(os.path.abspath(__file__))
 
 PACKAGE_NAME = "vllm_flash_attn"
 
-BASE_WHEEL_URL = (
-    "https://github.com/Dao-AILab/flash-attention/releases/download/{tag_name}/{wheel_name}"
-)
-
-# FORCE_BUILD: Force a fresh build locally, instead of attempting to find prebuilt wheels
-# SKIP_CUDA_BUILD: Intended to allow CI to use a simple `python setup.py sdist` run to copy over raw files, without any cuda compilation
-FORCE_BUILD = True
-SKIP_CUDA_BUILD = False
-# For CI, we want the option to build with C++11 ABI since the nvcr images use C++11 ABI
-FORCE_CXX11_ABI = torch._C._GLIBCXX_USE_CXX11_ABI
-
-
-def get_platform():
-    """
-    Returns the platform name as used in wheel filenames.
-    """
-    if sys.platform.startswith("linux"):
-        return f'linux_{platform.uname().machine}'
-    elif sys.platform == "darwin":
-        mac_version = ".".join(platform.mac_ver()[0].split(".")[:2])
-        return f"macosx_{mac_version}_x86_64"
-    elif sys.platform == "win32":
-        return "win_amd64"
-    else:
-        raise ValueError("Unsupported platform: {}".format(sys.platform))
-
+cmdclass = {}
+ext_modules = []
 
-def get_cuda_bare_metal_version(cuda_dir):
-    raw_output = subprocess.check_output([cuda_dir + "/bin/nvcc", "-V"], universal_newlines=True)
-    output = raw_output.split()
-    release_idx = output.index("release") + 1
-    bare_metal_version = parse(output[release_idx].split(",")[0])
+# We want this even if SKIP_CUDA_BUILD because when we run python setup.py sdist we want the .hpp
+# files included in the source distribution, in case the user compiles from source.
+subprocess.run(["git", "submodule", "update", "--init", "csrc/cutlass"])
```

Comment:
Maybe a TODO for later but we could replace this with the same FetchContent method as the vllm CMakeLists?